src/data_gradients/datasets/detection/coco_detection_dataset.py

```diff
@@ -0,0 +1,46 @@
+from typing import Union
+
+from data_gradients.datasets.detection.coco_format_detection_dataset import COCOFormatDetectionDataset
+
+
+class COCODetectionDataset(COCOFormatDetectionDataset):
+    """Coco Detection Dataset expects the exact same annotation files and dataset structure os the original Coco dataset.
+
+    #### Expected folder structure
+    The dataset folder structure should
+
+    Example:
+    ```
+    dataset_root/
+        ├── images/
+        │   ├── train2017/
+        │   ├── val2017/
+        │   └── ...
+        └── annotations/
+            ├── instances_train2017.json
+            ├── instances_val2017.json
+            └── ...
+    ```
+
+    #### Instantiation
+    To instantiate a dataset object for training data of the year 2017, use the following code:
+
+    ```python
+    from data_gradients.datasets.detection import COCODetectionDataset
+
+    train_set = COCODetectionDataset(root_dir="<path/to/dataset_root>", split="train", year=2017)
+    val_set = COCODetectionDataset(root_dir="<path/to/dataset_root>", split="val", year=2017)
+    ```
+    """
+
+    def __init__(self, root_dir: str, split: str, year: Union[int, str] = 2017):
```

Comment:
I don't think the `year` argument here is the right thing to do.
This may be ok for academic COCO dataset, but if someone exports dataset from Roboflow or any other external platform in COCO format, the notion of the year can be easily missing.
However the class constructor expects this input.
I think we should change this API to take only `images_dir` and `annotation_json` and that's it. Even root dir looks redundant to me, tbh.